### ISSUE TITLE
Fails when aggregating records

### ISSUE THREAD

**Issue by sadikovi** [sadikovi/spark-netflow#26]
When running aggregation with saving result in Parquet, package fails with error:

``` shell
java.lang.IllegalArgumentException: Failed to read record: -1 != 1
    at com.github.sadikovi.netflowlib.Buffers$ScanRecordBuffer$1.next(Buffers.java:172)
    at com.github.sadikovi.netflowlib.Buffers$ScanRecordBuffer$1.next(Buffers.java:118)
    at scala.collection.convert.Wrappers$JIteratorWrapper.next(Wrappers.scala:43)
    at com.github.sadikovi.spark.rdd.NetFlowFileRDD$$anonfun$compute$1$$anon$1.next(NetFlowFileRDD.scala:145)
    at com.github.sadikovi.spark.rdd.NetFlowFileRDD$$anonfun$compute$1$$anon$1.next(NetFlowFileRDD.scala:143)
    at scala.collection.Iterator$JoinIterator.next(Iterator.scala:194)
    at com.github.sadikovi.spark.rdd.NetFlowFileRDD$$anon$2.next(NetFlowFileRDD.scala:168)
    at com.github.sadikovi.spark.rdd.NetFlowFileRDD$$anon$2.next(NetFlowFileRDD.scala:166)
    at scala.collection.Iterator$$anon$11.next(Iterator.scala:370)
    at org.apache.spark.sql.execution.aggregate.TungstenAggregationIterator.processInputs(TungstenAggregationIterator.scala:372)
    at org.apache.spark.sql.execution.aggregate.TungstenAggregationIterator.start(TungstenAggregationIterator.scala:622)
    at org.apache.spark.sql.execution.aggregate.TungstenAggregate$$anonfun$doExecute$1.org$apache$spark$sql$execution$aggregate$TungstenAggregate$$anonfun$$executePartition$1(TungstenAggregate.scala:110)
    at org.apache.spark.sql.execution.aggregate.TungstenAggregate$$anonfun$doExecute$1$$anonfun$2.apply(TungstenAggregate.scala:119)
    at org.apache.spark.sql.execution.aggregate.TungstenAggregate$$anonfun$doExecute$1$$anonfun$2.apply(TungstenAggregate.scala:119)
    at org.apache.spark.rdd.MapPartitionsWithPreparationRDD.compute(MapPartitionsWithPreparationRDD.scala:64)
    at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:300)
    at org.apache.spark.rdd.RDD.iterator(RDD.scala:264)
    at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
    at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:300)
    at org.apache.spark.rdd.RDD.iterator(RDD.scala:264)
    at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:73)
    at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:41)
    at org.apache.spark.scheduler.Task.run(Task.scala:88)
    at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:214)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:745)
```

Query that was run:

``` scala
val df = sqlContext.read.format("com.github.sadikovi.spark.netflow").option("version", "5").load("file:/data/NetFlow/*/ft*")
val ad = df.groupBy($"srcip", $"dstip", $"srcport", $"dstport", $"protocol").agg(sum($"octets").as("octets"), sum($"packets").as("packets"))
ad.write.parquet("hdfs:/tmp/netflow-test-stringify.parquet")
```

**Comment by sadikovi** [comment]
Note that codegen version of package was used, but it seems like there is a bug in NetFlow library.

**Comment by sadikovi** [comment]
Sounds like regression, after predicate pushdown has been introduced.

**Comment by sadikovi** [comment]
It is a regression, it is a simple bug of `ReadAheadInputStream`. Addressed in #27
